torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -143,66 +144,140 @@ class ProcessGroupNCCL : public ProcessGroup {
   // this class in getFuture API of WorkNCCL. This Future is mostly a
   // wrapper to synchronize streams appropriately and it mostly enables
   // the async programming model of CUDA while trying to adhere to the
-  // Future interface.
+  // Future interface. FutureNCCL does not support NCCL_BLOCKING_WAIT flag
```

Comment:
Can you clarify this a bit? If I do run with NCCL_BLOCKING_WAIT, does this mean that if I call .wait() on a future returned by get_future() it still won't block? Would we want to add this support later on (seems like we would), if so can we file a GH issue? Same goes for `barrier()` - what wouldn't work if we did barrier() with FutureNCCL?

Reply:
I think in case of `NCCL_BLOCKING_WAIT=0` it will just wait until streams are synchronized and it will return before the whole NCCL kernel completed on the GPU. 
My understanding is that in case of `barrier()` and `NCCL_BLOCKING_WAIT, then callback will become even more complicated. Any thoughts @pritamdamania87, do we wanna support those later?